### PR TITLE
win_virtio_driver_update_by_installer: Add new test case

### PR DESCRIPTION
**Pull request by menli820**
Support update virtio-win drivers from previous virtio-win-guest-tools.exe
or previous virtio-win.iso.

ID: 1898762, 1898763
Signed-off-by: Menghuan Li menli@redhat.com

### CODE REVIEW

**Comment by menli820** [comment]
python3 ConfigTest.py --testcase=win_virtio_driver_update_by_installer --guestname=Win8..0 --platform=x86_64 --customsparams="cdrom_virtio = isos/windows/virtio-win-1.9.14-4.el8.iso" --machines=q35 --clone=no 

 (1/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.win_virtio_driver_update_by_installer.q35: PASS (888.12 s)
 (2/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.win_virtio_driver_update_by_installer.from_old_installer.q35: PASS (479.69 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2020-11-18T20.57-81d23a9/results.html
JOB TIME   : 1368.82 s


python3 ConfigTest.py --testcase=win_virtio_driver_update_by_installer --guestname=Win2016--platform=x86_64 --customsparams="cdrom_virtio = isos/windows/virtio-win-1.9.14-4.el8.iso" --machines=q35 --clone=no 

 (1/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_update_by_installer.q35: PASS (960.92 s)
 (2/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_update_by_installer.from_old_installer.q35: PASS (420.62 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2020-11-18T06.05-06bb719/results.html
JOB TIME   : 1382.52 s

**Comment by menli820** [comment]
@lijinlijin  please help to review, thanks

File: qemu/tests/win_virtio_driver_update_by_installer.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+INSTALL_TIMEOUT = 360
+OPERATION_TIMEOUT = 120
+
+
+def _pnpdrv_info(session, name_pattern, props=None):
+    cmd = wmic.make_query("path win32_pnpsigneddriver",
+                          "DeviceName like '%s'" % name_pattern,
+                          props=props, get_swch=wmic.FMT_TYPE_LIST)
+    return wmic.parse_list(session.cmd(cmd, timeout=QUERY_TIMEOUT))
+
```
Comment (lijinlijin) [inline qemu/tests/win_virtio_driver_update_by_installer.py:21]: same function with https://github.com/autotest/tp-qemu/pull/2482
We should reuse code

File: qemu/tests/cfg/win_virtio_driver_update_by_installer.cfg [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    use_mem_mem1 = no
+    share_mem = yes
+    guest_numa_nodes = shm0
+    numa_memdev_shm0 = mem-mem1
+    numa_nodeid_shm0 = 0
+    fs_binary_extra_options = " -o cache=auto"
+    variants:
+        - @default:
+        - from_old_installer:
+            update_from_previous_installer = yes
```
Comment (lijinlijin) [inline qemu/tests/cfg/win_virtio_driver_update_by_installer.cfg:82]: 1. whole cfg  is quite similar to https://github.com/autotest/tp-qemu/pull/2482 cfg file.
I suggest use same cfg. just different .py file(via type = xxx)
2. I think we should not use default for upgrade from iso file, for customers, upgrade from installer is their common usage

File: qemu/tests/win_virtio_driver_update_by_installer.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        """
+        change iso for virtio-win
+        :param cdrom_virtio: iso file
+        """
+        virtio_iso = utils_misc.get_path(data_dir.get_data_dir(),
+                                         cdrom_virtio)
+        logging.info("Changing virtio iso image to '%s'" % virtio_iso)
+        vm.change_media("drive_virtio", virtio_iso)
+
+    def get_driver_inf_path(session, driver_name):
```
Comment (lijinlijin) [inline qemu/tests/win_virtio_driver_update_by_installer.py:52]: same function with #2482
We should reuse code

File: qemu/tests/win_virtio_driver_update_by_installer.py [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        inf_middle_path = ("{name}\\{arch}" if media_type == "iso"
+                           else "{arch}\\{name}").format(name=guest_name,
+                                                         arch=guest_arch)
+        inf_find_cmd = 'dir /b /s %s\\%s.inf | findstr "\\%s\\\\"'
+        inf_find_cmd %= (viowin_ltr, driver_name, inf_middle_path)
+        inf_path = session.cmd(inf_find_cmd, timeout=OPERATION_TIMEOUT).strip()
+        logging.info("Found inf file '%s'", inf_path)
+        return inf_path
+
+    def uninstall_driver(session, driver_name, device_name, device_hwid):
```
Comment (lijinlijin) [inline qemu/tests/win_virtio_driver_update_by_installer.py:86]: same function with #2482
We should reuse code

**Comment by lijinlijin** [comment]
@menli820 
There are too much dup with https://github.com/autotest/tp-qemu/pull/2482
I think you need to re-struct this patch to reduce the dup code and resuse some common function/steps.

**Comment by menli820** [comment]
close for re_structure it to https://github.com/autotest/tp-qemu/pull/2482/

**Pull request closed without merging**